### PR TITLE
CP-14775 Remove obsolete pv-drivers-version-checking

### PR DESCRIPTION
**Pull request by thomassa**

### CODE REVIEW

**Comment by thomassa** [comment]
Obviously the three commits should be squashed, but I think the review might be easier with them separate.

File: ocaml/xapi/xapi_pv_driver_version.ml [hunk 1]
```diff
@@ -176,7 +46,7 @@ let get_drivers_version os_version drivers_version =
 		(* in rolling upgrade rio slaves will not put micro vsn in database, but we musn't report
 			 "Unknown", since then is_ok_for_migrate check will fail... *)
 		let micro = lookup_driver_key_with_default "micro" (-1) in
-		(* added in Orlando *)
+		(* added in Orlando *) (* XXX: linux guest agent doesn't report build number while all windows ones do *)
```
Comment (thomassa) [inline ocaml/xapi/xapi_pv_driver_version.ml]: Comment about linux agent is copied from deleted line 150

**Comment by johnelse** [comment]
This does indeed look like it does what the CP ticket called for :)

**Comment by thomassa** [comment]
Thank you @johnelse -- shall squash and merge.

**Pull request merged**